src/transformations.jl

```diff
@@ -97,22 +121,69 @@ julia> StatsBase.transform(dt, X)
  -1.0   0.0  1.0
 ```
 """
-function fit(::Type{ZScoreTransform}, X::AbstractArray{<:Real,2}; center::Bool=true, scale::Bool=true)
-    d, n = size(X)
-    n >= 2 || error("X must contain at least two columns.")
+function fit(::Type{ZScoreTransform}, X::AbstractMatrix{<:Real};
+             dims::Union{Integer,Nothing}=nothing, center::Bool=true, scale::Bool=true)
+    if dims == 1
+        n, l = size(X)
+        n >= 2 || error("X must contain at least two rows.")
+        m, s = mean_and_std(X, 1)
+    elseif dims == 2
+        l, n = size(X)
+        n >= 2 || error("X must contain at least two columns.")
+        m, s = mean_and_std(X, 2)
+    elseif dims === nothing
+        Base.depwarn("fit(t, x) is deprecated: use fit(t, x, dims=2) instead", :fit)
+    end
+    T = eltype(X)
+    return ZScoreTransform(l, dims, (center ? vec(m) : zeros(T, 0)),
+                                    (scale ? vec(s) : zeros(T, 0)))
+end
+
+function fit(::Type{ZScoreTransform}, X::AbstractVector{<:Real};
+             dims::Union{Integer,Nothing}=nothing, center::Bool=true, scale::Bool=true)
+    if dims == nothing
+        Base.depwarn("fit(t, x) is deprecated: use fit(t, x, dims=2) instead", :fit)
+    elseif dims != 1
+        throw(DomainError(dims, "fit only accept dims=1 over a vector. Try fit(t, x, dims=1)."))
+    end
 
     T = eltype(X)
-    m, s = mean_and_std(X, 2)
+    m, s = mean_and_std(X)
+    return ZScoreTransform(1, dims, (center ? [m] : zeros(T, 0)),
+                                    (scale ? [s] : zeros(T, 0)))
+end
+
+function transform!(y::AbstractMatrix{<:Real}, t::ZScoreTransform, x::AbstractMatrix{<:Real})
+    l = t.len
+    size(x,2) == size(y,2) == l || throw(DimensionMismatch("Inconsistent dimensions."))
+    n = size(y,1)
+    size(x,1) == n || throw(DimensionMismatch("Inconsistent dimensions."))
 
-    return ZScoreTransform(d, (center ? vec(m) : zeros(T, 0)),
-                              (scale ? vec(s) : zeros(T, 0)))
+    m = t.mean
+    s = t.scale
+
+    if isempty(m)
+        if isempty(s)
+            if x !== y
+                copyto!(y, x)
+            end
+        else
+            broadcast!(/, y, x, s')
+        end
+    else
+        if isempty(s)
+            broadcast!(-, y, x, m')
+        else
+            broadcast!((x,m,s)->(x-m)/s, y, x, m', s')
+        end
+    end
+    return y
 end
 
-function transform!(y::AbstractVecOrMat{<:Real}, t::ZScoreTransform, x::AbstractVecOrMat{<:Real})
-    d = t.dim
-    size(x,1) == size(y,1) == d || throw(DimensionMismatch("Inconsistent dimensions."))
-    n = size(y,2)
-    size(x,2) == n || throw(DimensionMismatch("Inconsistent dimensions."))
+function transform!(y::AbstractVector{<:Real}, t::ZScoreTransform, x::AbstractVector{<:Real})
```

Comment:
Rather than duplicating the function for `x::AbstractVector`, call `reshape(x, :, 1)` and call `transform!` on it to reuse the matrix method. Same below.